src/main/java/com/liveramp/captain/example/ExampleCaptainWorkflow.java

```diff
@@ -0,0 +1,312 @@
+import com.liveramp.captain.request_submitter.RequestSubmitter;
+import com.liveramp.captain.status.CaptainStatus;
+import com.liveramp.captain.status_retriever.StatusRetriever;
+import com.liveramp.captain.step.CaptainStep;
+import com.liveramp.captain.waypoint.AsyncWaypoint;
+import com.liveramp.captain.waypoint.ControlFlowWaypoint;
+import com.liveramp.captain.waypoint.SyncWaypoint;
+import com.liveramp.commons.collections.map.MapBuilder;
+import com.liveramp.daemon_lib.Daemon;
+
+public class ExampleCaptainWorkflow {
+  static final String APP1 = "App1";
+  static final String APP2 = "App2";
+  static final String STEP1 = "Step1";
+  static final String STEP2 = "Step2";
+  static final String STEP3 = "Step3";
+  static final String STEP4 = "Step4";
+  static final String STEP5 = "DONE";
+
+  private Logger LOG = LoggerFactory.getLogger(ExampleCaptainWorkflow.class);
+
+  /**
+   * For sake of example let's say we have our team is responsible for running some job that performs analytics on a file.
+   * <p>
+   * Doing this requires 3 main steps:
+   * 1) Wait for ingestion of the file to complete by some other team (i.e. poll until we know the file is ready for us to use.
+   * 2) Kick off a map reduce job that's handled by some service.
+   * 3) When that completes, report the information from the map reduce job to some other team's service. In this example,
+   * we'll have a split pipeline, where depending on the file, we report the information to the Data Science Team 1's
+   * Service or to Data Science Team 2's Service.
+   * <p>
+   * When thinking about this with Captain, the fact that we have two distinct pipelines should hint that we want to
+   * use a "MultiApp" system (or perhaps optional steps, but for this example we're not going to do that).
+   * <p>
+   * In this class, you will see a skeleton version of how you might implement this use case using captain.
+   */
+  public static void main(String[] args) throws IllegalAccessException, IOException, InstantiationException {
+    new ExampleCaptainWorkflow().build().start();
+  }
+
+  private Daemon<CaptainRequestConfig> build() throws IOException, IllegalAccessException, InstantiationException {
+    // thin persistence layer for sake of example that looks something like one might store a request in their own db.
+    final ExampleInternals.MockDb mockDb = new ExampleInternals.MockDb();
+
+    // Remember we need to provide Captain 3 main resources: 1) Manifest, 2) Config Producer, 3) Request Updater
+
+    // 1) Manifest
+    // Since we have have two apps running in our captain instance. We provide captain with two manifests and a key
+    // to determine manifest it should use for each request. Hopefully by skimming over these manifests it's pretty easy
+    // to map what they're doing at a high level to our stated goals above.
+    final Manifest exampleManifestApp1 = new DefaultManifestImpl(Lists.newArrayList(
```

Comment:
hmm. i was thinking of calling that out in the readme more. the approach i took here is how do you built a captain app with the least resistance possible. left out things like worrying about locks. i think you're probably right though that it would be good to call out that it's a builtin and you can implement your own.